### ISSUE TITLE
Create new splits when moving windows towards an edge, like standard vim

### ISSUE THREAD

**Issue by BenjaminBrandtner** [simeji/winresizer#17]
Check out this short video: I've got three horizontal splits, all on top of each other and I use the standard vim window commands (`CTRL-W` + `H`,`J`,`K`,`L`) to move one of them around.

![ezgif-4-055b8fcd6194](https://user-images.githubusercontent.com/37274870/49029016-fbb97a00-f1a3-11e8-83da-f0690f34d3d9.gif)

When I try to move any of the windows to the right or left, it creates a new vertical split that is completely filled by the window I moved.
And when I try to move the vertically split window downwards or upwards, it deletes the vertical split and becomes a horizontal split again.

Basically, standard vim behaviour is: Trying to move a window towards the "edge" of the entire vim window changes the type of split it is. While the standard window movement commands can be pretty unintutive, this is a feature I quite like.

Any chance you could replicate it in the `window move mode` of this plugin?
